DOCS-715: servo models and configuration refactor (#1148)
viamrobotics/docs: docs/components/servo/_index.md

description: "A hobby servo is a special type of small motor whose position you can precisely control."
tags: ["servo", "components"]
icon: "/components/img/components/servo.svg"
no_list: true
# SME: #team-bucket
---

> A *servo* is a ["RC" or "hobby" servo motor](https://learn.adafruit.com/adafruit-motor-selection-guide/rc-servos).

Reviewer: Either
```suggestion
The servo component supports ["RC" or "hobby" servo motors](https://learn.adafruit.com/adafruit-motor-selection-guide/rc-servos).
```
or
```suggestion
A *servo* is an ["RC" or "hobby" servo motor](https://learn.adafruit.com/adafruit-motor-selection-guide/rc-servos).
```
I think the former sounds less tautological.

Reviewer 2: I like the first one

description: "A hobby servo is a special type of small motor whose position you can precisely control."
tags: ["servo", "components"]
icon: "/components/img/components/servo.svg"
no_list: true
# SME: #team-bucket
---

{{% alert title="Tip" color="tip" %}}

> The Viam servo component supports [hobby servos](https://learn.adafruit.com/adafruit-motor-selection-guide/rc-servos).

Reviewer 2: I think keeping the info about `built-in closed-loop control` would be good. And I think this first sentence may go outside, and above the tip. I'm thinking information that is required standard info should probably always be in the normal text rather than in a tip? Might be worth thinking about a bit more though

Author: Yeah, I was struggling with phrasing this if including more info about the encoder/position feedback to differentiate servo & encoded motor configuration.

From what I can tell the "built-in closed-loop control" isn't inaccurate but also isn't necessarily the best phrasing if people reading this don't already have a good understanding of open-loop vs closed-loop control systems, and if they do then they're probably more interested in understanding what's happening in that system? @randhid do you have any thoughts on phrasing here?

Reviewer 3: I agree this is more friendly to people who don't understand these concepts or care about them, and if people do then they are likely more interested in our motor APIs. If we expand our servo apis, then this may change but we'll cross that bridge if it happens.

> If you have another type of servomotor, it is probably coupled with an [encoder](/components/encoder), not potentiometer, for position feedback.

Reviewer 2: so are hobby motors always with a potentiometer? Would it be correct to say The Viam servo component supports hobby servos that have potentiometers?

Author: @npentrel I *think* that is the case, but should confirm with @randhid-- Rand, is this distinction accurate? and if so do you think the sensor specifics for "Viam servos" is important info to include?

Reviewer 3: hobby *servos* that we support in both drivers always have a potentiometer, have a potentiometer. Motors can have many encoder types. There are non-pin based servos that exist with encoders that we do not support yet. This sentence is very specific to `gpio` and `pi` which are the only ones that exist in rdk.

Check your device's data sheet and configure these servos as [encoded motors](/components/motor/gpio/encoded-motor).

{{% /alert %}}

A *servo* is a [radio-control (RC) or hobby servo motor](https://learn.adafruit.com/adafruit-motor-selection-guide/rc-servos).
These are small motors with built-in potentiometer position sensors, enabling you to control the angular position of the servo precisely.

> As servos can use a lot of power, drawing voltage away from a [board](/components/board), you should power your servo with its own power supply.

Reviewer 3: actually, I always recommend a separate power supply since we have more boards than the pi now, and the current draw from different boards is harder to reason about. It is a cleaner recommendation, even if it is a bit inconvenient.

But if you'd like to explicitly separate the cheaper blue rcs and the beefier gpio servos, that might be the best thing to do here.

> As servos can use a lot of power, drawing voltage away from a [board](/components/board), you should power your servo with its own power supply.

Reviewer 2: ```suggestion
As servos can use a lot of power, drawing voltage away from a [board](/components/board), you should power your servo with its own power supply in most cases.
```

Could add something like that to allow for the alternative?

> As servos can use a lot of power, drawing voltage away from a [board](/components/board), you should power your servo with its own power supply.

Author: @npentrel done!

> Example wiring diagram for a hobby servo wired to a [`pi` board](/components/board/pi):

Reviewer 2: ```suggestion
The following shows an example wiring diagram for a hobby servo wired to a [`pi` board](/components/board/pi):
```

- A power supply for the board
- A power supply for the servo

## Configuration

{{% alert title="Tip" color="tip" %}}

The Viam servo component supports [hobby servos](https://learn.adafruit.com/adafruit-motor-selection-guide/rc-servos).

> If you have another type of servo motor, it is probably coupled with an [encoder](/components/encoder), not potentiometer, for position feedback.

Reviewer: ```suggestion
If your motor is coupled with an [encoder](/components/encoder), not a potentiometer, for position feedback, you should not configure it as a servo.
```

> If you have another type of servo motor, it is probably coupled with an [encoder](/components/encoder), not potentiometer, for position feedback.

Reviewer: ```suggestion
If your motor is coupled with an [encoder](/components/encoder), not a potentiometer, for position feedback, you should not configure it as a servo.
```

> Check your device's data sheet and configure these servos as [encoded motors](/components/motor/gpio/encoded-motor).

Reviewer: ```suggestion
Check your device's data sheet and configure that type of servo as an [encoded motor](/components/motor/gpio/encoded-motor).
```

- A power supply for the board
- A power supply for the servo

{{% alert title="Note" color="note" %}}

> Instead of powering the servo with a separate power supply, you may choose to power it using the 5V and ground pins on the board.

Reviewer: Last time we talked about it I was told to add this in because most people will actually do it this way.
